src/utils.ts

```diff
@@ -142,3 +143,106 @@ export function voidFn() {}
 export function assertUnreachable(x: never): never {
   throw new Error("Didn't expect to get here");
 }
+
+/**
+ * A custom React Hook for calculating whether a thumb overlaps
+ * another and whether labels could/should merge.
+ * @param rangeRef - React ref value of Range component
+ * @param values - current Range values Array
+ * @param index - thumb index
+ * @param separator - string to separate thumb values
+ */
+export const useThumbOverlap = (
+  rangeRef: IRangeRef | undefined,
```

Comment:
```suggestion
  rangeRef?: IRangeRef,
```

Reply:
Also, if this can be undefined, should it be the last param?

Reply:
The hook doesn't really work without it but the ref value can be `null` the first time the hook is run. The hook won't do anything without the ref value.